cmake/app/boilerplate.cmake

```diff
@@ -193,6 +193,12 @@ endif()
 assert(BOARD "BOARD not set")
 message(STATUS "Board: ${BOARD}")
 
+include(${ZEPHYR_BASE}/board_aliases.cmake OPTIONAL)
+if(${BOARD}_BOARD_ALIAS)
+  set(BOARD_ALIAS ${BOARD} CACHE STRING "Board alias, provided by user")
+  set(BOARD ${${BOARD}_BOARD_ALIAS})
```

Comment:
Will this work for overlays?

Reply:
If the overlay is named using the canonical name it should.  It won't find the overlay if you name it using the alias. IMO it shouldn't; the alias is just for personal convenience.